packages/cinterop/src/darwin/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -339,6 +339,22 @@ actual object RealmInterop {
         }
     }
 
+    actual fun realm_config_set_should_compact_on_launch_function(
+        config: NativePointer,
+        callback: CompactOnLaunchCallback
+    ) {
+        realm_wrapper.realm_config_set_should_compact_on_launch_function(
+            config.cptr(),
+            staticCFunction<COpaquePointer?, uint64_t, uint64_t, Boolean> { userdata, total, used ->
```

Comment:
Maybe also add a comment that this is leaking with a reference to the core issue